markdown/java-jvm/collections.md

```diff
@@ -196,6 +196,24 @@ Beispiel /hash_example
 * `Object.hashCode()` wird genutzt, um einen Hash-Wert für ein Objekt zu berechnen und in Hash-Containern zu speichern/suchen.
 * Der `hashCode`-Vertrag definiert die Verhaltensweisen von `hashCode()`.
 
+## Überblick
+
+|Collection|Sorted|allow duplicate|Synchronized|allow `null`|Einfügen|Löschen|Get|
+|----------|------|---------------|------------|------------|--------|-------|------|
+|`LinkedList`|ja|ja|nein|ja|O(1)|O(n)|O(n)|
+|`ArrayList`|ja|ja|nein|ja|O(n)|O(n)|O(1)|
```

Comment:
Hier hab ich etwas Bauchweh:
Das Einfügen in einer ArrayList kann O(1) sein, wenn im Array noch platz ist. Wenn das Array vergrößert (und daher kopiert werden muss) ist es O(n).
Ich weiß das BC in ADS immer vom "worst-case" Ausgeht, wenn es um Komplexitätsberechnung geht. 
Dann müsste man aber Konsequenter weise aber auch bei HashMaps von O(n) ausgehen. Aber ich kann doch niemanden erzählen, dass das suchen/einfügen/löschen in einer HashMap die Komplexität O(n) hat außer ich führe das konkret aus. Und dafür fehlt Zeit/Platz an dieser Stelle.

Reply:
fussnoten gehen in tabellen leider nicht. aber man kann ja mit `notes` einen kommentar direkt unter der tabelle einbauen. (oder hier konkret auch ohne `notes`, da die tabelle eh im nicht-folien-bereich ist, oder?)

im prinzip ist das immer worst-case, aber für den theorie-fall. du hast jetzt wegen der konkreten implementierung beim einfügen des 11. elements die kopiersituation, aber die spielt nach meiner einschätzung im o-kalkül keine rolle. du könntest das array ja auch größer anlegen von vornherein, oder?

was sagt denn die javadoc im jdk? haben die da o-kalkül-angaben?

Reply:
Hab es auf O(1) geändert und eine Note eingebaut.

Reply:
Good point. Bin auch nur vom "normalen" Einfügen ausgegangen und das wäre in meinen Augen das Hinzufügen am Ende.

Reply:
naja, wenn man ne linked-list im kopf hat... aber sonst ist doch einfügen != anhängen.

Reply:
ja.

ich hab nur noch nie was anderes mit einer arraylist gemacht.

Ich pass die Tabelle nochmal an, damit solche Spezifikationen auch dargestellt werden können,

Reply:
aye.

üblicherweise schreibt man immer "am anfang", "am ende", "irgendwo in der mitte" (siehe https://en.wikipedia.org/wiki/Dynamic_array).

Reply:
hab ich eingebaut.
Da markdown keine "Unterspalten" unterstützt, hab ich eine Numbers Table erstellt und ein Screenshot eingebunden. Vorlage ist auch mit im Repo

Reply:
keks.